### PR TITLE
Caching orchestration + other improvements

### PR DESCRIPTION
**Pull request by YiannisGkoufas**
Bringing caching orchestration logic
Keys generation within the pod
Kubeflow samples

### CODE REVIEW

File: src/dataset-operator/cmd/manager/main.go [hunk 1]
```diff
@@ -126,6 +128,14 @@ func main() {
...
 		os.Exit(1)
 	}
+	if err := cephv1.AddToScheme(mgr.GetScheme()); err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+	if err := objectstorage.AddToScheme(mgr.GetScheme()); err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
```
Comment (christian-pinto) [inline src/dataset-operator/cmd/manager/main.go:140]: Why is rook ceph referenced in the main? This should be part of the caching plugin, please check this is fixed throughout this PR.

**Review by christian-pinto — CHANGES_REQUESTED** [review]

File: src/dataset-operator/cmd/manager/main.go (hunk 1 above)
Comment (YiannisGkoufas) [reply src/dataset-operator/cmd/manager/main.go:140]: Good catch, the logic/imports for the ceph plugin should not be on the main dataset operator

Comment (YiannisGkoufas) [reply src/dataset-operator/cmd/manager/main.go]: Check the last commit https://github.com/IBM/dataset-lifecycle-framework/pull/39/commits/a20c83c7934ed1e6f5fed082521e2edef333a21a

Comment (christian-pinto) [reply src/dataset-operator/cmd/manager/main.go]: Perfect, thanks!

**Comment by christian-pinto** [comment]
This project has received funding from the European Union’s Horizon 2020 research and innovation programme under grant agreement No 825061, H2020 Evolve.

**Pull request merged**